bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -472,6 +447,35 @@ protected synchronized void deactivate(ComponentContext componentContext) {
         readyService.unregisterTracker(this);
     }
 
+    private void thingUpdated(final Thing thing) {
+        thingUpdatedLock.add(thing.getUID());
+        AccessController.doPrivileged((PrivilegedAction<@Nullable Void>) () -> {
+            final Provider<Thing> provider = thingRegistry.getProvider(thing);
+            if (provider == null) {
+                throw new IllegalArgumentException(MessageFormat.format(
+                        "Provider for thing {0} cannot be determined because it is not known to the registry",
+                        thing.getUID().getAsString()));
+            }
+            if (provider instanceof ManagedProvider) {
+                @SuppressWarnings("unchecked")
+                final ManagedProvider<Thing, ThingUID> managedProvider = (ManagedProvider<Thing, ThingUID>) provider;
+                managedProvider.update(thing);
+            } else {
+                logger.debug("Only updating thing {} in the registry because provider {} is not managed.",
+                        thing.getUID().getAsString(), provider);
+                final Thing oldThing = thingRegistry.get(thing.getUID());
+                if (oldThing == null) {
+                    throw new IllegalArgumentException(
+                            MessageFormat.format("Cannot update thing {0} because it is not known to the registry",
+                                    thing.getUID().getAsString()));
+                }
+                thingRegistry.updated(provider, oldThing, thing);
+            }
+            return null;
+        });
+        thingUpdatedLock.remove(thing.getUID());
+    }
+
```

Comment:
While reviewing I found that the `Message.format` seven lines below contain an NBSP after the first placeholder. Can you fix that, if you are near this code anyway? Thanks.